mixer/pkg/runtime/config/yaml/reader.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yaml
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+	"github.com/pkg/errors"
+)
+
+// Callback defines the callback made by the Reader.
+type Callback interface {
```

Comment:
I suggest we rename interface to `Visitor` and method to `Visit`

Walk( visitor ) is common naming pattern.